tests/ci/sqltest.py

```diff
@@ -0,0 +1,156 @@
+
+    stopwatch = Stopwatch()
+
+    temp_path = TEMP_PATH
+    reports_path = REPORTS_PATH
+
+    check_name = sys.argv[1]
+
+    if not os.path.exists(temp_path):
+        os.makedirs(temp_path)
+
+    pr_info = PRInfo()
+
+    gh = Github(get_best_robot_token(), per_page=100)
+    commit = get_commit(gh, pr_info.sha)
+
+    rerun_helper = RerunHelper(commit, check_name)
+    if rerun_helper.is_already_finished_by_status():
+        logging.info("Check is already finished according to github status, exiting")
+        sys.exit(0)
+
+    docker_image = get_image_with_version(reports_path, IMAGE_NAME)
+
+    build_name = get_build_name_for_check(check_name)
+    print(build_name)
+    urls = read_build_urls(build_name, reports_path)
+    if not urls:
+        raise Exception("No build URLs found")
+
+    for url in urls:
+        if url.endswith("/clickhouse"):
+            build_url = url
+            break
+    else:
+        raise Exception("Cannot find the clickhouse binary among build results")
+
+    logging.info("Got build url %s", build_url)
+
+    workspace_path = os.path.join(temp_path, "workspace")
+    if not os.path.exists(workspace_path):
+        os.makedirs(workspace_path)
+
+    run_command = get_run_command(
+        pr_info.number, pr_info.sha, build_url, workspace_path, docker_image
+    )
+    logging.info("Going to run %s", run_command)
+
+    run_log_path = os.path.join(temp_path, "run.log")
+    with open(run_log_path, "w", encoding="utf-8") as log:
+        with subprocess.Popen(
+            run_command, shell=True, stderr=log, stdout=log
+        ) as process:
+            retcode = process.wait()
+            if retcode == 0:
+                logging.info("Run successfully")
+            else:
+                logging.info("Run failed")
+
+    subprocess.check_call(f"sudo chown -R ubuntu:ubuntu {temp_path}", shell=True)
+
+    check_name_lower = (
+        check_name.lower().replace("(", "").replace(")", "").replace(" ", "")
+    )
+    s3_prefix = f"{pr_info.number}/{pr_info.sha}/sqltest_{check_name_lower}/"
+    paths = {
+        "run.log": run_log_path,
+        "main.log": os.path.join(workspace_path, "main.log"),
+        "server.log.zst": os.path.join(workspace_path, "server.log.zst"),
+        "server.err.log.zst": os.path.join(workspace_path, "server.err.log.zst"),
+        "report.html": os.path.join(workspace_path, "report.html"),
+        "test.log": os.path.join(workspace_path, "test.log"),
+    }
+
+    s3_helper = S3Helper()
+    for f in paths:
+        try:
+            paths[f] = s3_helper.upload_test_report_to_s3(paths[f], s3_prefix + f)
+        except Exception as ex:
+            logging.info("Exception uploading file %s text %s", f, ex)
+            paths[f] = ""
+
+    report_url = GITHUB_RUN_URL
+    if paths["report.html"]:
+        report_url = paths["report.html"]
+
+    status = "success"
+    description = "See the report"
+    test_result = TestResult(description, "OK")
```

Comment:
add logs and report files here

Reply:
Ok. I want to make a small working prototype first.
This check has limited use. It should exist; everything else we will address later (or never).